### ISSUE TITLE
npm install --production and npm shrinkwrap do not behave consistently

### ISSUE THREAD

**Issue by afrasso** [npm/npm#13430]
[shrinkwrap-all.json.txt](https://github.com/npm/npm/files/372619/shrinkwrap-all.json.txt)
[shrinkwrap-all-plus-dev.json.txt](https://github.com/npm/npm/files/372621/shrinkwrap-all-plus-dev.json.txt)
[shrinkwrap-production-only.json.txt](https://github.com/npm/npm/files/372620/shrinkwrap-production-only.json.txt)
#### I'm opening this issue because:
- [ ] npm is crashing.
- [ ] npm is producing an incorrect install.
- [X] npm is doing something I don't understand.
- [ ] Other (_see below for feature requests_):
#### What's going wrong?

It seems that npm install --production seems to identify and exclude transitive dev dependencies correctly, but npm shrinkwrap does not.
#### How can the CLI team reproduce the problem?

Here are the dependencies in my package.json file:

```
"dependencies": {
  "grunt": "^1.0.1",
  "grunt-npm-helper": "0.0.2"
},
"devDependencies": {
  "grunt-contrib-jshint": "^1.0.0",
  "grunt-jscs": "^3.0.0",
  "grunt-mocha-istanbul": "^5.0.1",
  "grunt-mocha-test": "^0.12.7",
  "istanbul": "^0.4.4",
  "mocha": "^2.5.3"
}
```

When clear out my node_modules folder and run `npm install --production`, the dev dependencies do not get installed. When I run `npm shrinkwrap`, only these dependencies are listed (see shrinkwrap-production-only.json).

If I again clear out my node_modules folder and run `npm install`, the dev dependencies _do_ get installed. When I run `npm shrinkwrap`, it seems that all installed modules are listed except for the dev dependencies explicitly identified in the package.json file (see shrinkwrap-all.json). Note that the packages "align-text" and "amdefine" near the top of the file are present in shrinkwrap-all.json, but not shrinkwrap-production-only.json, and note that istanbul is not listed in either file.

Finally, if I run `npm shrinkwrap --dev`, all modules are listed including explicit dev dependencies (see shrinkwrap-all-plus-dev.json). Note that the only differences between shrinkwrap-all.json and shrinkwrap-production-only.json are the 6 explicitly identified dev dependencies listed in package.json.
### supporting information:
- `npm -v` prints:
  
  `C:\>npm -v`
  `3.8.8`
- `node -v` prints:
  
  `C:\>node -v`
  `v5.9.1`
- `npm config get registry` prints: It's a private artifactory repository, but I doubt that's related.
- Windows, OS X, or Linux?: Windows
- Network issues:
  - Geographic location where npm was run: US Northeast
  - [ ] I use a proxy to connect to the npm registry.
  - [ ] I use a proxy to connect to the web.
  - [ ] I use a proxy when downloading Git repos.
  - [ ] I access the npm registry via a VPN
  - [ ] I don't use a proxy, but have limited or unreliable internet access.
- Container:
  - [ ] I develop using Vagrant on Windows.
  - [ ] I develop using Vagrant on OS X or Linux.
  - [ ] I develop / deploy using Docker.
  - [ ] I deploy to a PaaS (Triton, Heroku).

Thanks for the help!

**Comment by Kovensky** [comment]
I believe this will be fixed by #13399

**Comment by shanetmoore** [comment]
#13399 has been closed and this one is still in place. 

Installing dev dependencies even with production flags set is pretty inefficient. Is there an updated plan for this?
